### PR TITLE
HPCC-18240 Roxie not handling out-of-sequence UDP messages well

### PR DESCRIPTION
**Pull request by richardkchapman**
Delete code that can never be useful (as server code to track resendable data
is disabled) but can cause performance issues and other problems on systems
where udp gets out of sequence.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [ ] Security
  - [x] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
TBD

### CODE REVIEW

File: roxie/udplib/udpsha.hpp [hunk 1]
```diff
@@ -173,64 +171,45 @@ class simple_queue
...
-        hdr.lastSequenceSeen = 0;
-        hdr.missingCount = 0;
+        length = cmd = destNodeIndex = max_data = 0;
 #ifdef CRC_MESSAGES
-        hdr.crc = calcCRC();
+        crc = calcCRC();
 #endif
     }
 
     UdpPermitToSendMsg(const UdpPermitToSendMsg &from)
```
Comment (ghalliday) [inline roxie/udplib/udpsha.hpp]: minor: could use =default;

**Comment by richardkchapman** [comment]
Basically just deleting tons of code that serves no purpose but may (a) hold spinlocks for long periods and (b) get very confused by out-of-sequence UDP packets

**Comment by ghalliday** [comment]
As far as I can see the changes look good.

File: roxie/udplib/udpsha.hpp [hunk 2]
```diff
@@ -29,6 +29,11 @@ typedef bool (*PKT_CMP_FUN) (void *pkData, void *key);
 
 #define UDP_SEQUENCE_COMPLETE 0x80000000
 
+// Flag bits in pktSeq field
+#define UDP_PACKET_COMPLETE           0x80000000
+#define UDP_PACKET_RESERVED           0x40000000  // Not used - could move UDP_SEQUENCE_COMPLETE here?
+#define UDP_PACKET_SEQUENCE_MASK     ~0x3fffffff
```
Comment (ghalliday) [inline roxie/udplib/udpsha.hpp]: @richardkchapman I don't think the ~ is correct

Comment (richardkchapman) [reply roxie/udplib/udpsha.hpp]: No. Kept changing my mind where the ~ belonged!

**Comment by richardkchapman** [comment]
@mckellyln Can you confirm that the testing is complete and successful (may be worth repeating with latest commit, to be on the safe side)

**Pull request merged**